components/logins/README.md

```diff
@@ -15,11 +20,69 @@ The relevant directories are as follows:
   editing logins using the code in `src`.
 - `ffi`: The Rust public FFI bindings. This is a (memory-unsafe, by necessity)
   API that is exposed to Kotlin and Swift. It leverages the `ffi_support` crate
-  to avoid many issues and make it more safe than it othrwise would be. At the
+  to avoid many issues and make it more safe than it otherwise would be. At the
   time of this writing, it uses JSON for marshalling data over the FFI, however
   in the future we will likely use protocol buffers.
 - `android`: This contains android bindings to logins, written in Kotlin. These
   use JNA to call into to the code in `ffi`.
 - `ios`: This contains the iOS binding to logins, written in Swift. These use
   Swift's native support for calling code written in C to call into the code in
   `ffi`.
+
+## Features
+1. Locally encrypted storage of username and password information
+1. Create, Update and Delete (CRUD) operations for login data
+1. Syncing of logins for devices and apps connected via Firefox Accounts
+1. Import functionality from existing login storage (ex: Fx Desktop or Fennec)
+1. Data migration functionality from Fennec to Firefox Preview storage
+
+
+## Business Logic
+
+### Record storage
+
+At any given time records can exist in 3 places, the local storage, the remote record, and the shared parent.  The shared parent refers to a record that has been synced previously and is referred to in the code as the mirror.
+Login records are encrypted and stored locally. For any record that does not have a shared parent the login component tracks that the record has never been synced.
+
+Reference the [Logins chapter of the synconomicon](https://mozilla.github.io/application-services/synconomicon/ch01.1-logins.html) for detailed information on the record storage format.
+
+### Sign-out behavior
+When the user signs out of their Firefox Account, we reset the storage and clear the shared parent.
+
+### Merging records
+When records are added, the logins component performs a three-way merge between the local record, the remote record and the shared parent (last update on the server).  Details on the merging algorithm are contained in the [generic sync rfc](https://github.com/mozilla/application-services/blob/1e2ba102ee1709f51d200a2dd5e96155581a81b2/docs/design/remerge/rfc.md#three-way-merge-algorithm).
```

Comment:
Oooh, nice, I think @thomcc's doc is an excellent framework for understanding how merging works! We may want to clarify which fields get merged how (for example, that we use `take_max` for `timeLastUsed` and `sum` for `timesUsed`), but that doesn't need to be done here.